cpp/oneapi/dal/algo/decision_forest/common.hpp

```diff
@@ -50,33 +65,60 @@ using v1::by_default;
 
 namespace v1 {
 
+///
+/// Available identifiers to specify the variable importance mode
+///
 enum class variable_importance_mode {
-    none, /* Do not compute */
-    mdi, /* Mean Decrease Impurity.
-                       Computed as the sum of weighted impurity decreases for all nodes where the variable is used,
-                       averaged over all trees in the forest */
-    mda_raw, /* Mean Decrease Accuracy (permutation importance).
-                       For each tree, the prediction error on the out-of-bag portion of the data is computed
-                       (error rate for classification, MSE for regression).
-                       The same is done after permuting each predictor variable.
-                       The difference between the two are then averaged over all trees. */
-    mda_scaled /* Mean Decrease Accuracy (permutation importance).
-                       This is MDA_Raw value scaled by its standard deviation. */
+    /// Do not compute variable importance
+    none,
+
+    /// Mean Decrease Impurity.
+    /// Computed as the sum of weighted impurity decreases for all nodes where the variable is used,
+    /// averaged over all trees in the forest
+    mdi,
+    /// Mean Decrease Accuracy (permutation importance).
+    /// For each tree, the prediction error on the out-of-bag portion of the data is computed
+    /// (error rate for classification, MSE for regression).
+    /// The same is done after permuting each predictor variable.
+    /// The difference between the two are then averaged over all trees.
+    mda_raw,
+
+    /// Mean Decrease Accuracy (permutation importance).
+    /// This is MDA_Raw value scaled by its standard deviation.
+    mda_scaled
 };
 
+///
+/// Available identifiers to specify the error metric mode
+///
 enum class error_metric_mode : std::uint64_t {
+    /// Do not compute error metric
     none = 0x00000000ULL,
+    /// Train produces $1 \\times 1$ table with cumulative prediction error for out of bag observations
     out_of_bag_error = 0x00000001ULL,
+    /// Train produces $n \\times 1$ table with prediction error for out of bag observations
     out_of_bag_error_per_observation = 0x00000002ULL
 };
 
+///
+/// Available identifiers to specify the infer mode
+///
 enum class infer_mode : std::uint64_t {
-    class_labels = 0x00000001ULL, /*!< Numeric table of size n x 1 with the predicted labels >*/
-    class_probabilities =
-        0x00000002ULL /*!< Numeric table of size n x p with the predicted class probabilities for each observation >*/
+    /// Infer produces a $n \\times 1$  table with the predicted labels
+    class_labels = 0x00000001ULL,
+    /// Infer produces $n \\times c$ table with the predicted class probabilities for each observation
+    class_probabilities = 0x00000002ULL
 };
 
-enum class voting_mode { weighted, unweighted };
+///
+/// Available voting modes for averaging trees predictions
+///
```

Comment:
Do we need extra `///` on top and bottom? Does clang-format require it? As far as I remember, coding guidelines does not require it

Reply:
fixed